src/python_testing/matter_testing_infrastructure/chip/testing/apps.py

```diff
@@ -102,3 +104,92 @@ def __init__(self, app: str, rpc_server_port: int, storage_dir: str,
+            ota_image_path: Path to a file containing an OTA image (cannot be used with image_list_path)
+            image_list_path: Path to a file containing a list of OTA images (cannot be used with ota_image_path)
+            extra_args: Additional command line arguments
+        """
+
+        if extra_args is None:
+            extra_args = []
+
+        if ota_image_path and image_list_path:
+            raise ValueError("Cannot specify both ota_image_path and image_list_path")
+
+        if not ota_image_path and not image_list_path:
+            raise ValueError("Must specify either ota_image_path or image_list_path")
+
+        # Build OTA-specific arguments
+        ota_args = []
+        if ota_image_path:
+            ota_args.extend(["--filepath", ota_image_path])
+        elif image_list_path:
+            ota_args.extend(["--otaImageList", image_list_path])
+
+        combined_extra_args = ota_args + extra_args
+
+        # Initialize with the combined arguments
+        super().__init__(app=app, storage_dir=storage_dir, discriminator=discriminator,
+                         passcode=passcode, port=port, extra_args=combined_extra_args)
+
+    def create_acl_entry(self, dev_ctrl, provider_node_id: int, requestor_node_id: Optional[int] = None):
+        """Create ACL entries to allow OTA requestors to access the provider.
+
+        Args:
+            dev_ctrl: Device controller for sending commands
+            provider_node_id: Node ID of the OTA provider
+            requestor_node_id: Optional specific requestor node ID for targeted access
+
+        Returns:
+            Result of the ACL write operation
+        """
+        # Standard ACL entry for OTA Provider cluster
+        acl_entries = [
+            {
+                "fabricIndex": 1,
+                "privilege": Clusters.AccessControl.Enums.AccessControlEntryPrivilegeEnum.kAdminister,
+                "authMode": Clusters.AccessControl.Enums.AccessControlEntryAuthModeEnum.kCase,
+                "subjects": [dev_ctrl.nodeId] if hasattr(dev_ctrl, 'nodeId') else [self.DEFAULT_ADMIN_NODE_ID],
+                "targets": None
+            },
+            {
+                "fabricIndex": 1,
+                "privilege": Clusters.AccessControl.Enums.AccessControlEntryPrivilegeEnum.kOperate,
+                "authMode": Clusters.AccessControl.Enums.AccessControlEntryAuthModeEnum.kCase,
+                "subjects": [requestor_node_id] if requestor_node_id else None,
+                "targets": [{"cluster": Clusters.OtaSoftwareUpdateProvider.id, "endpoint": None, "deviceType": None}]
+            }
+        ]
+
+        return dev_ctrl.WriteAttribute(
+            nodeid=provider_node_id,
+            attributes=[(0, Clusters.AccessControl.id, Clusters.AccessControl.Attributes.Acl.attribute_id, acl_entries)]
+        )
+
+    def __del__(self):
+        # Override to safely handle kvs_fd that might not exist if constructor failed
```

Comment:
who creates a kvs_fd? Why is this a file descriptor and not a python file object?